src/fpm/manifest.f90

```diff
@@ -126,6 +126,8 @@ subroutine get_package_data(package, file, error, apply_defaults)
             end if
         end if
 
+        call add_fpm_is_windows_macro(package%preprocess)
```

Comment:
I know `cargo` supports [identifying the OS and selectively compiling the code](https://github.com/rust-lang/cargo/blob/a9465fe7515d2ef61c82035baf01cb968fb4eba0/Cargo.toml#L91), and it seems like `fpm` can do something similar with this PR. (`fpm` also knows the operating system type when it compiles the code.)